be1-go/hub/organizer.go

```diff
@@ -590,21 +579,7 @@ func (c *laoChannel) processMessageObject(public message.PublicKey, data message
 			}
 		}
 
-		c.inboxMu.Lock()
-		msg, ok := c.inbox[msgEncoded]
-		if !ok {
-			// TODO: We received a witness signature before the message itself.
-			// We ignore it for now but it might be worth keeping it until we
-			// actually receive the message
-			log.Printf("failed to find message_id %s for witness message", msgEncoded)
-			c.inboxMu.Unlock()
-			return nil
-		}
-		msg.WitnessSignatures = append(msg.WitnessSignatures, message.PublicKeySignaturePair{
-			Witness:   public,
-			Signature: witnessData.Signature,
-		})
-		c.inboxMu.Unlock()
+		c.inbox.addWitnessSignature(witnessData.MessageID, public, witnessData.Signature)
```

Comment:
We're not checking the return value of `addWitnessSignature` here. Is that safe?

Reply:
No I don't think this is safe! I'm gonna add a check!